src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -872,23 +884,59 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
             public override void Initialize(AnalysisContext context)
             {
                 context.RegisterCompilationAction(compilationContext =>
-                    compilationContext.ReportDiagnostic(CodeAnalysis.Diagnostic.Create(UnsupportedDescriptor, Location.None)));
+                {
+                    try
+                    {
+                        ThrownException = null;
+                        compilationContext.ReportDiagnostic(CodeAnalysis.Diagnostic.Create(UnsupportedDescriptor, Location.None));
+                    }
+                    catch (Exception e)
+                    {
+                        ThrownException = e;
+                        throw;
+                    }
+                });
             }
         }
 
         [Fact, WorkItem(4376, "https://github.com/dotnet/roslyn/issues/4376")]
         public void TestReportingDiagnosticWithInvalidId()
         {
             string source = @"";
+            CSharpCompilation compilation = CreateCompilationWithMscorlib45(source);
```

Comment:
Please use `CreateCompilation` unless you have a specific reason for targeting net45. The goal is to eliminate these legacy helpers.

Reply:
This is preserved unchanged from the original code (leaving conversion for a separate work item). New tests added by this pull request already use `CreateCompilation`.